dlc-manager/src/manager.rs

```diff
@@ -827,6 +810,57 @@ where
         Ok(())
     }
 
+    /// Check for pending cooperative close transactions
+    fn check_pending_close_transactions(&self) -> Result<(), Error> {
+        // Get all Confirmed contracts that might have pending close transactions
+        for contract in self.store.get_confirmed_contracts()? {
+            // Skip channel contracts (they have their own monitoring)
+            if contract.channel_id.is_some() {
+                continue;
+            }
+
+            // Check each pending close transaction
+            for pending_close_tx in &contract
+                .accepted_contract
+                .dlc_transactions
+                .pending_close_txs
+            {
+                let confirmations = self
+                    .blockchain
+                    .get_transaction_confirmations(&pending_close_tx.compute_txid())?;
+
+                if confirmations >= NB_CONFIRMATIONS {
+                    // Found a fully confirmed pending close - move directly to Closed
+                    let pnl = contract.accepted_contract.compute_pnl(pending_close_tx)?;
+                    let closed_contract = ClosedContract {
+                        attestations: None, // Cooperative close has no attestations
+                        signed_cet: None,   // Cooperative close doesn't use a CET
```

Comment:
Why set it to None, I think it's fine to save the closing transaction here, unless it breaks something?

Reply:
You're right, signed_cet can be the closing tx, which is must more elegant. Updated in [9d84b29](https://github.com/p2pderivatives/rust-dlc/pull/255/commits/9d84b29b372bc2c7486a4127e0c538bbe6294a27)